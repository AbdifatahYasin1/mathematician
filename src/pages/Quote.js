import React from 'react';

const Quote = () => (
  <div id="quote">
    <p>
      Mathematics is the language
      in which God has written the universe.
    </p>
    <cite>- Galileo Galilei</cite>
  </div>
);

export default Quote;
