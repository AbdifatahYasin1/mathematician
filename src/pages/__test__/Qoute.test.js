// test qoute
import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

describe('Testing Qoute', () => {
  test('should render correctly', () => {
    render(<Quote />);
  });
});
test('should have a length of 1', () => {
  const { getAllByText } = render(<Quote />);
  const text = getAllByText(/Mathematics is the study of numbers, shapes, and patterns./);
  expect(text).toHaveLength(1);
});
it('fisrt snap shot test', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
