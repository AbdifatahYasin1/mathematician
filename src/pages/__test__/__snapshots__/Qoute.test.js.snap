// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fisrt snap shot test 1`] = `
<div
  id="quote"
>
  <p>
    Mathematics is the language in which God has written the universe.
  </p>
  <cite>
    - Galileo Galilei
  </cite>
</div>
`;
